api/src/main/java/marquez/db/migrations/V44_1__BackfillAirflowParentRuns.java

```diff
@@ -0,0 +1,193 @@
+          AND event->'run'->'facets'->'airflow_version' IS NOT NULL
+          ) e ON e.run_uuid=r.uuid
+      WHERE e.parent_run_id IS NOT NULL
+""";
+  public static final String INSERT_PARENT_JOB_QUERY =
+      """
+      INSERT INTO jobs (uuid, type, created_at, updated_at, namespace_uuid, name, description,
+      namespace_name, current_location)
+      SELECT :uuid, type, created_at, updated_at, namespace_uuid, :name, description, namespace_name,
+      current_location
+      FROM jobs
+      WHERE namespace_name=:namespace AND name=:jobName
+      ON CONFLICT(name, namespace_uuid) WHERE parent_job_uuid IS NULL
+      DO UPDATE SET updated_at=now()
+      RETURNING uuid
+      """;
+  public static final String INSERT_PARENT_RUN_QUERY =
+      """
+      INSERT INTO runs (uuid, created_at, updated_at, current_run_state, external_id, namespace_name, job_name, job_uuid, location, transitioned_at, started_at, ended_at)
+      SELECT :parentRunUuid, created_at, updated_at, current_run_state, :externalRunid, :namespace, :jobName, :parentJobUuid, location, transitioned_at, started_at, ended_at
+      FROM runs
+      WHERE uuid=:runUuid
+      ON CONFLICT (uuid) DO NOTHING
+      """;
+
+  @Override
+  public void migrate(Context context) throws Exception {
+    Jdbi jdbi = Jdbi.create(context.getConnection());
+    List<ParentRun> parentRuns =
+        jdbi.withHandle(
+            h ->
+                h.createQuery(FIND_AIRFLOW_PARENT_RUNS_SQL)
+                    .map(
+                        rs -> {
+                          String parentRunId = rs.getColumn("parent_run_id", String.class);
+                          if (parentRunId == null) {
+                            return null;
+                          }
+                          String parentJobName = rs.getColumn("parent_job_name", String.class);
+                          String dagName =
+                              parentJobName.contains(".")
+                                  ? parentJobName.substring(0, parentJobName.lastIndexOf('.'))
+                                  : parentJobName;
+                          String parentJobNamespace =
+                              rs.getColumn("parent_job_namespace", String.class);
+                          String runUuid = rs.getColumn("run_uuid", String.class);
+                          log.info(
+                              "Found likely airflow run {} with parent {}.{} run {}",
+                              runUuid,
+                              parentJobNamespace,
+                              parentJobName,
+                              parentRunId);
+                          UUID parentRunUuid;
+                          try {
+                            parentRunUuid = UUID.fromString(parentRunId);
+                          } catch (IllegalArgumentException e) {
+                            parentRunUuid = Utils.toNameBasedUuid(dagName, parentRunId);
```

Comment:
Would it be more ideal to control flow by checking if `dagName` is `null` (or make it an optional), then via an exception?